src/js/DeviceInfoProvider.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react'
+import { isMobile, isMobileLandscape } from './breakpoints'
+
+export default class DeviceInfoProvider extends React.Component {
```

Comment:
You could use newer react hooks like useEffect here. New components in react-autoql should as we try to move this package frwd. You could simply ask a AI tool to help with the conversion